ush/fv3gfs_dwn_nems.sh

```diff
@@ -1,109 +1,84 @@
 #! /usr/bin/env bash
 
-# this script generates 0.25/0.5/1/2.5 deg pgb files for each small Grib file
-# Hui-Ya Chuang 01/2014: First Version
-# Fanglin Yang  09/2015: Modified to use WGRIB2 instead of COPYGB2 for interpolation
-# Fanglin Yang  02/2016: remove 0.5-deg and 2.5deg output to speed up post            
-# Fanglin Yang  09/11/2017: add option opt24 to turn on bitmap (from Wen Meng) 
-# Wen Meng 12/2017: add trim_rh.sh for triming RH values larger than 100.
-# Wen Meng 01/2018: add flag PGB1F for turning on/off wgrib1 pgb data at 1.00 deg. generation.
-# Wen Meng 02/2018: add flag PGBS for turning on/off pgb data at 1.0 and 0.5 deg. generation.
-# Wen Meng 10/2019: Use bilinear interpolation for LAND, It can trancate land-sea mask as 0 or 1.
-# Wen Meng 11/2019: Teak sea ice cover via land-sea mask.
+# This script takes in a master grib file and creates products at various interpolated resolutions
+# Generate 0.25 / 0.5 / 1 degree interpolated grib2 files for each input grib2 file
+# trim's RH and tweaks sea-ice cover
 
-source "$HOMEgfs/ush/preamble.sh"
 
-export tmpfile=$1
-export fhr3=$2
-export iproc=$3
-export nset=$4
+source "${HOMEgfs}/ush/preamble.sh"
 
-export CNVGRIB=${CNVGRIB:-${grib_util_ROOT}/bin/cnvgrib}
-export COPYGB2=${COPYGB2:-${grib_util_ROOT}/bin/copygb}
-export WGRIB2=${WGRIB2:-${wgrib2_ROOT}/bin/wgrib2}
-export TRIMRH=${TRIMRH:-$USHgfs/trim_rh.sh}
-export MODICEC=${MODICEC:-$USHgfs/mod_icec.sh}
+input_file=${1:-"pgb2file_in"}  # Input pressure grib2 file
+output_file_prefix=${2:-"pgb2file_out"}  # Prefix for output grib2 file; the prefix is appended by resolution e.g. _0p25
+grid_string=${3:-"0p25"}  # Target grids; e.g. "0p25" or "0p25:0p50"; If multiple, they need to be ":" seperated
 
-export opt1=' -set_grib_type same -new_grid_winds earth '
-export opt21=' -new_grid_interpolation bilinear  -if '
-export opt22=":(CSNOW|CRAIN|CFRZR|CICEP|ICSEV):"
-export opt23=' -new_grid_interpolation neighbor -fi '
-export opt24=' -set_bitmap 1 -set_grib_max_bits 16 -if '
-export opt25=":(APCP|ACPCP|PRATE|CPRAT):"
-export opt26=' -set_grib_max_bits 25 -fi -if '
-export opt27=":(APCP|ACPCP|PRATE|CPRAT|DZDT):"
-export opt28=' -new_grid_interpolation budget -fi '
-if [ $machine = "S4" ]; then
```

Comment:
can we alias `WGRIB2` to contain `-ncpu 1` such that semantically; `WGRIB2="wgrib2 -ncpu 1"

Reply:
It's been a while since I dug into this job, but I seem to recall there are some locations where wgrib2 is called outside of the MPMD subtasks.  For these, it would still be useful to run with all available CPUs.  If I'm mistaken about that and wgrib2 is only called within MPMD subtasks, then that would be a solid way to handle it.

Reply:
you are right.  There are instances where wgrib2 is called outside of mpmd subtasks.  This is true even in this PR where wgrib2 is being used to split up a larger file into pieces for mpmd sub-processing.
We can make a note of this and find an optimal solution when dealing with wgrib2 inside of mpmd processing at a later time, if that is acceptable.

Reply:
Yep, sounds good.  Thanks!